CA-388210: use unique datapaths for concurrent VDI copies

edwintorok opened the pull request:
Testing on a box shows that the new unique arguments for 'domain' work, but opening as a draft PR, because this needs wider testing in XenRT.

lindig commented on ocaml/xapi-storage-script/main.ml:
```diff
@@ -49,6 +49,23 @@ let backend_backtrace_error name args backtrace =
+  or vbd/domid/device.
+  For regular guests keep the domain as passed by XAPI (an integer).
+ *)
+let domain_of ~dp ~vm' =
```
Requiring a prime in named label is a bit awkward - is there a reason this is not just `~vm`?

edwintorok replied:
All the parameters where this function is called were called `vm'`, however renaming should be easy, I've added a commit at the beginning of this PR that renames `vm'` to `vm`, and then we can avoid having `'` in the labeled argument.

edwintorok commented:
I rebased and retargeted this to feature/perf and removed the commit that enables concurency (since that one still triggers a deadlock that we're trying to track down in the storage code).

lindig commented:
Can we merge this?
